examples/energy-gateway-app/commodity-metering/src/CommodityMeteringEventTriggers.cpp

```diff
@@ -0,0 +1,249 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <CommodityMeteringInstance.h>
+#include <app/clusters/commodity-metering-server/CommodityMeteringTestEventTriggerHandler.h>
+#include <array>
+#include <cstdint>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::CommodityMetering;
+
+namespace MeteredQuantitySamples {
+// Define component arrays as constexpr
+namespace Sample1 {
+static constexpr uint32_t TariffComponents1[] = { 0x5001, 0x5003 };
+static constexpr uint32_t TariffComponents2[] = { 0x5002, 0x5003, 0x5004 };
+
+// Non-constexpr storage for the actual data
+static const Structs::MeteredQuantityStruct::Type Data[] = {
+    { DataModel::List<const uint32_t>(TariffComponents1, sizeof(TariffComponents1) / sizeof(TariffComponents1[0])), 3500 },
+    { DataModel::List<const uint32_t>(TariffComponents2, sizeof(TariffComponents2) / sizeof(TariffComponents2[0])), -2000 }
+};
+} // namespace Sample1
+
+namespace Sample2 {
+static constexpr uint32_t TariffComponents1[] = { 0x6001 };
+static constexpr uint32_t TariffComponents2[] = { 0x6002, 0x6003 };
+
+static const Structs::MeteredQuantityStruct::Type Data[] = {
+    { DataModel::List<const uint32_t>(TariffComponents1, sizeof(TariffComponents1) / sizeof(TariffComponents1[0])), 4200 },
+    { DataModel::List<const uint32_t>(TariffComponents2, sizeof(TariffComponents2) / sizeof(TariffComponents2[0])), -1500 }
+};
+} // namespace Sample2
+} // namespace MeteredQuantitySamples
+
+namespace {
+
+class OldCommodityMeteringAttributes
+{
+private:
+    Instance * mInstance = nullptr;
+
+    DataModel::Nullable<DataModel::List<Structs::MeteredQuantityStruct::Type>> mMeteredQuantity;
+    DataModel::Nullable<uint32_t> mMeteredQuantityTimestamp;
+    Globals::TariffUnitEnum mTariffUnit;
+
+    DataModel::List<Structs::MeteredQuantityStruct::Type> GetMeteredQuantityDataSample(uint8_t presetIdx)
+    {
+        switch (presetIdx)
+        {
+        case 0:
+            return DataModel::List<Structs::MeteredQuantityStruct::Type>(
+                const_cast<Structs::MeteredQuantityStruct::Type *>(MeteredQuantitySamples::Sample1::Data),
```

Comment:
If we have const_cast involved, we are doing something wrong.

Concretely, should this be returning `DataModel::List<const Structs::MeteredQuantityStruct::Type>`?